packages/tokenbank_v1/contracts/modules/transfers/TokenBalances.sol

```diff
@@ -35,22 +35,20 @@ contract TokenBalances is BaseModule
     }
 
     function tokenBalance(
-        address wallet,
         address token
         )
         public
         view
         returns (uint)
     {
         if (token == address(0)) {
-            return wallet.balance;
+            return address(this).balance;
         } else {
-            return ERC20(token).balanceOf(wallet);
+            return ERC20(token).balanceOf(address(this));
```

Comment:
@Brechtpd , my understanding is that since this method is bound to a wallet, so the "address(this)" should be the wallet.

Reply:
`address(this)` is the address of the `TokenBalances` module. `transact` uses `call` and not `delegatecall` so there is nothing special with how the functions are called even through the wallet.

Reply:
then several static methods will have to be updated.